### PR TITLE
reorient board function

### PR DESCRIPTION
**Pull request by sojeri**
does nothing on left/right direction

turns board 90&deg; on up/down direction :)

### CODE REVIEW

File: javascripts/2048.js [hunk 1]
```diff
@@ -32,21 +32,57 @@ function moveTile(tile, direction) {
...
 
-Board.protoype.update(direction) {
+console.log(board.board);
+
+Board.prototype.move = function(direction) {
   // this is the movement function
-  // var reorientedBoard = this.split();
-  // this.condense();
-}
+  var reorientedBoard = this.split(direction);
```
Comment (corinnepingul) [inline javascripts/2048.js]: does this need to be `this.reorient(direction)` now?

File: javascripts/2048.js [hunk 2]
```diff
@@ -32,21 +32,57 @@ function moveTile(tile, direction) {
...
+// board.reorient() reorients the board into arrays based on direction
+Board.prototype.reorient = function(direction) {
+  var method;
+
+  if (direction == "left" || direction == "right")
+    method = "leftReorient";
+  else // "up" || "down"
+    method = "downReorient";
+
+  return this[method].call(this); // execute the method in the current context
```
Comment (corinnepingul) [inline javascripts/2048.js:62]: oooh, nice!

Comment (corinnepingul) [inline javascripts/2048.js:62]: I should use this in the method I'm writing now...

File: javascripts/2048.js [hunk 3]
```diff
@@ -32,21 +32,57 @@ function moveTile(tile, direction) {
...
+};
+
+Board.prototype.leftReorient = function() {
+  return this.board; // or do we want to modify the board in place?
+};
+
+Board.prototype.downReorient = function() {
+  var reorientedBoard = [];
+
+  for (var oldCol = 0; oldCol < 4; oldCol++) {
```
Comment (corinnepingul) [inline javascripts/2048.js]: Should we make some kind of constant variable `boardLength` that holds the value 4 so we can reference it throughout this file? I'm not sure the exact syntax for doing it in js, but I'm sure we can figure it out.

File: javascripts/2048.js (hunk 1 above)
Comment (sojeri) [inline javascripts/2048.js]: ....yes

File: javascripts/2048.js [hunk 4]
```diff
@@ -32,21 +32,57 @@ function moveTile(tile, direction) {
...
+    method = "downReorient";
+
+  return this[method].call(this); // execute the method in the current context
+};
+
+Board.prototype.leftReorient = function() {
+  return this.board; // or do we want to modify the board in place?
+};
+
+Board.prototype.downReorient = function() {
```
Comment (corinnepingul) [inline javascripts/2048.js]: can we have these function names also refer to the right and up directions? Something like `leftRightReorient` and `downUpReorient`? I know it's messy, but I got a bit confused at first when looking at these functions.

File: javascripts/2048.js (hunk 3 above)
Comment (sojeri) [inline javascripts/2048.js]: yes! I will add it to the Board constructor. :)

File: javascripts/2048.js [hunk 5]
```diff
@@ -32,21 +32,57 @@ function moveTile(tile, direction) {
...
+
+board = new Board([
+  [2,     2,   0,   0], // [2,   4,  16, 512],
+  [4,     8,   4,   2], // [2,   8,  32, 256],
+  [16,   32, 128,  64], // [0,   4, 128, 128],
+  [512, 256, 128,  32] //  [0,   2,  64,  32]
+]);
 
-Board.protoype.update(direction) {
+console.log(board.board);
```
Comment (corinnepingul) [inline javascripts/2048.js]: do you still need the console.logs (this one and also the one on line 51)?

File: javascripts/2048.js (hunk 4 above)
Comment (sojeri) [inline javascripts/2048.js]: horizontal & vertical is a great idea! :)

File: javascripts/2048.js (hunk 5 above)
Comment (sojeri) [inline javascripts/2048.js]: nope, I forgot to remove them after testing. :)

**Pull request merged**